src/inmanta/data/__init__.py

```diff
@@ -5290,7 +5290,7 @@ async def create_for_partial_compile(
                 $4,
                 $5,
                 (
-                    SELECT array_agg(rid)
+                    SELECT coalesce(array_agg(rid), ARRAY[]::character varying[])
```

Comment:
This makes sure that no new NULL values will be added to the database, but it doesn't fix the NULL values that already exist in the database. Is this intentional?

Reply:
No, nice catch. Should I add a migration to transform NULL values into empty arrays ?
Should the migration also add a not null constraint ?

Reply:
> Should I add a migration to transform NULL values into empty arrays ?

Yes.

> Should the migration also add a not null constraint ?

Yes.